testing/trino-tests/src/test/java/io/trino/tests/TestInformationSchemaConnector.java

```diff
@@ -158,7 +158,7 @@ public void testMetadataCalls()
                 "SELECT count(*) from test_catalog.information_schema.tables WHERE table_name LIKE 'test_t_ble1'",
                 "VALUES 2",
                 new MetadataCallsCount()
-                        .withListSchemasCount(1)
+                        .withListSchemasCount(3)
```

Comment:
that looks like a regression, not an improvement

shouldn't `Return more generic schema filter if possible` commit contain some test changes showing that it's indeed an improvement, at least under some circumstances?
otherwise, a revert of that commit will look like an obvious win.

Reply:
I think `Return more generic schema filter if possible` it is more of an early optimization that could introduce some sort of regression for queries with `like` applied on `schemaName` by invoking `listSchemaNames` multiple times. If the number of schemas to load crosses 100 we pass on all schema as an input i.e defaultPrefixes(catalogName) so which should allow us to rely on new batch API. WDYT ? cc: @kokosing , @findepi , @huberty89

Reply:
discussed this offline with @huberty89